File: App/Resources/AlgorithmDetails/selectionsort/selectionsort.rb
def sort(arr)
  n = arr.length
  i = 0
  while i < n - 1
    min_idx = i
    j = i + 1
    while j < n
      if arr[j] < arr[min_idx]
        min_idx = j
      end
      j += 1
    end
    arr[min_idx], arr[i] = arr[i], arr[min_idx]
    i += 1
  end
end

array = [0, 39, 21, 62, 91, 77, 14, 23,
  90, 69, 51, 81, 68, 83, 32, 56]
sort(array)
p array

File: App/Resources/AlgorithmDetails/selectionsort/selectionsort.java
import java.util.Arrays;

public class selectionsort {
  public static void sort(int[] arr) {
    selectionSort(arr);
  }

  public static void selectionSort(int[] arr) {
    int n = arr.length;
    for (int i = 0; i < n - 1; i++) {
      int minIdx = i;
      for (int j = i + 1; j < n; j++) {
        if (arr[j] < arr[minIdx]) {
          minIdx = j;
        }
      }
      int temp = arr[minIdx];
      arr[minIdx] = arr[i];
      arr[i] = temp;
    }
  }

  public static void main(String[] args) {
    int[] array = {
      0, 39, 21, 62, 91, 77, 14, 23,
      90, 69, 51, 81, 68, 83, 32, 56
    };
    sort(array);
    System.out.println(Arrays.toString(array));
  }
}
